clis/teliod/src/main.rs

```diff
@@ -0,0 +1,240 @@
+//! Main and implementation of config and commands for Teliod - simple telio daemon for Linux and OpenWRT
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use nix::libc::SIGTERM;
+use signal_hook_tokio::Signals;
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::{Path, PathBuf},
+    process::id,
+    str::FromStr,
+};
+use tracing::{error, info, level_filters::LevelFilter};
+
+mod comms;
+mod daemon;
+
+use crate::{comms::DaemonSocket, daemon::Daemon};
+
+use telio::{device::Device, telio_model::features::Features, telio_utils::select};
+
+use serde::{de, Deserialize, Deserializer, Serialize};
+
+use futures::stream::StreamExt;
+
+struct TeliodDaemon;
+
+impl Daemon for TeliodDaemon {
+    const NAME: &'static str = "teliod";
+
+    fn get_wd_path() -> Result<PathBuf> {
+        if Path::new("/run").exists() {
+            Ok(PathBuf::from("/run/"))
+        } else {
+            Ok(PathBuf::from("/var/run/"))
```

Comment:
I guess we should check that `/var/run` exists as well, and return error if nor `/run` nor `/var/run` exists.